web-client/src/ducks/requests/functions.ts

```diff
@@ -75,3 +76,23 @@ export const setUserRequest = async ({
     .doc(requestId)
     .withConverter(RequestFirestoreConverter)
     .set(requestPayload);
+
+export const updateUserRequest = async ({
+  requestId,
+  fieldsToUpdate,
+}: IupdateRequest) => {
+  const data: any = { ...fieldsToUpdate };
+  // remove undefined values
+  Object.keys(data).forEach(key => data[key] === undefined && delete data[key]);
+
+  const timeNow = firebase.firestore.Timestamp.now();
+
+  data.updatedAt = firebase.firestore.Timestamp.now();
+  data.pinRating && (data.pinRatedAt = timeNow);
+  data.cavRating && (data.cavRatedAt = timeNow);
+
+  return firestore
+    .collection('requests')
+    .doc(requestId)
+    .update({ ...data });
+};
```

Comment:
@ashwinkjoseph As I mentioned, I'm not happy with this implementation. However, it works well. I intended to use `setRequest` action but had issues with the request converter I guess.

The issue is simply, we cannot use `toFirebase` with the request stored in redux. Because in the converter `toFirebase` method calls `toObject` and it looks for path variable from a reference, i.e `pinUserRef.path` and it gets undefined because of pinUserRef stored as a string i.e `users/someuid`. 

I'm looking forward to hearing your opinions here. As soon as you mention, I'll work on changes :)

Reply:
have you been using the instance of the request class to modify this?
if not that could be the problem

Reply:
Fixed 8af5a85, now it's good to go